### PR TITLE
Add support for long-option aliases.

### LINKED ISSUE #7048: Add alias support for long-options strings
**Description of the desired feature**

Add alias support for long-options strings. Entries for long-options in the common or module-specific .h files may be modified to add aliases for the triggering --whatever strings, or the long-directive or long-modifier strings via use of the | character, e.g., "region|reg|domain".

**Are you willing to help implement and maintain this feature?** Yes

### PR DESCRIPTION
**Pull request by rbdavis**
**Description of proposed changes**

Add support for long-option aliases.

Fixes #7048

**Reminders**

- [ ] Make sure that your code follows our style. Use the other functions/files as a basis.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Describe changes to function behavior and arguments in a comment below the function declaration.
- [ ] If adding new functionality, add a detailed description to the documentation and/or an example.

### CODE REVIEW

File: src/gmt_common_longoptions.h [hunk 1]
```diff
@@ -52,12 +66,12 @@
     {   0, 'B', "axis",
                 "x,y,z",                 "x,y,z",
                 "a,f,l,L,p,s,S,u",       "angle,fancy,label,hlabel,prefix,alt_label,alt_hlabel,unit" },
-    {   0, 'J', "projection",
+    {   0, 'J', "projection|proj",
                 "",                      "",
                 "d,a,t,v,w,z,f,k,r",     "d,a,t,v,w,z,f,k,r" },
-    {   0, 'R', "region",
+    {   0, 'R', "region|reg|domain",
```
Comment (joa-quim) [inline src/gmt_common_longoptions.h:72]: ``reg`` clashes with ``registration`` and is not a good alias for -R. Julia uses ``region`` or ``limits`` (like Matlab's xlim, ylim, etc). Why introducing ``domain``?

File: src/longopt/grdimage_inc.h [hunk 2]
```diff
@@ -42,7 +42,7 @@ static struct GMT_KEYWORD_DICTIONARY module_kw[] = { /* Local options for this m
                   "a,d,m,n",           "azimuth,default,ambient,intensity" },
 	{ 0, 'M', "monochrome",        "", "", "", "" },
 	{ 0, 'N', "noclip",            "", "", "", "" },
-	{ 0, 'Q', "transvalue",
+	{ 0, 'Q', "transvalue|alphacolor",
```
Comment (joa-quim) [inline src/longopt/grdimage_inc.h:45]: ``transvalue`` is not nice. Again, Julia uses ``alpha_color``, ``nan_alpha``, PyGMT use ``nan_transparent``.

File: src/longopt/grdinterpolate_inc.h [hunk 3]
```diff
@@ -37,7 +37,7 @@ static struct GMT_KEYWORD_DICTIONARY module_kw[] = { /* Local options for this m
 	{ 0, 'S', "pointseries",
 	          "",                      "",
 	          "h",                     "header" },
-	{ 0, 'T', "timeinc",
+	{ 0, 'T', "timeinc|inc|range",
```
Comment (joa-quim) [inline src/longopt/grdinterpolate_inc.h:40]: ``timeinc`` is associated with time but -T is for any type of increments.

Comment (rbdavis) [reply src/longopt/grdinterpolate_inc.h:40]: All of the initial entries in the above alias lists were the pre-existing ones approved earlier. The aliases I added were either pre-approved by Paul or added by me after looking at Julia and trying to add compatibility.

Personally I do not really care about any of these that much. The rest of you can discuss and I will make whatever changes Paul directs me to make.

**Review by joa-quim — APPROVED** [review]
Better commit this and change later than live it stalled.

**Pull request merged**